### ISSUE TITLE
Improve CI with regards to data repo updates

### ISSUE THREAD

**Issue by DanielKotik** [mala-project/mala#537]
Every time we update the data repo, we have to change something in multiple places, two places in the RODARE URL, then the version of the zip file and on top adapt the `mv` command as well. This is not very convenient, we should try to improve on that in the future. For now, I would leave it as is.

**Comment by DanielKotik** [comment]
One of the annoying issues is that `test-data-x.y.z.zip` contains a top-level directory with an unknown suffix (a commit id) that needs to be renamed to what the CI expects in subsequent steps, namely `mala_data`:

https://github.com/mala-project/mala/blob/64fe38b75d7d93310278d3e6a4ac1946d5f30056/.github/workflows/cpu-tests.yml#L185-L186

Ideally, `unzip` should be able to unzip without the root folder into a custom folder. This seems, however, not possible as is outlined [here](https://unix.stackexchange.com/questions/38162/how-can-i-force-unzip-zip-not-to-create-a-subdirectory-when-i-extract-it).

But `tar` [allows](https://superuser.com/questions/899684/how-to-untar-an-archive-without-the-root-folder-but-keeping-the-full-hierarchy) for such a skipping of the root folder.

```sh
❯ tree -L 1 test-data-46a6992
test-data-46a6992
├── Be2
├── LICENSE
├── linking_tester.npy
├── README.md
├── test_repo.py
└── workflow_test

# create compressed tar archive
❯ tar -czvf test-data-1.8.0.tar.gz test-data-46a6992

# extract compressed tar archive into custom folder while skipping root directory
❯ mkdir mala_data && tar -xzvf test-data-1.8.0.tar.gz --strip-components=1 -C mala_data

❯ tree -L 1 mala_data
mala_data
├── Be2
├── LICENSE
├── linking_tester.npy
├── README.md
├── test_repo.py
└── workflow_test
```

This would allow the code lines 185-186 to be replaced by `mkdir mala_data && tar -xzvf test-data-1.8.0.tar.gz --strip-components=1 -C mala_data` without having to change the suffix of the root directory as well as the data archive version (`x.y.z`).

**Comment by RandomDefaultUser** [comment]
I agree that having the commit hash in the name of the folder, and then having to go through the hassle of changing all these constants for every update is annoying. However, "Would it be possible to switch from zip to tar.gz format in the data repo?" is probably not possible. RODARE automatically downloads a .zip file from GH, we don't have any influence on that. What we _could_ do is ask them if they can adapt their workflow. It shouldn't be too hard, since a release comes both with a zip and a tar...

**Comment by RandomDefaultUser** [comment]
We could of course also switch to Zenodo. I am not familiar enough to know if there are quota/size restriction in Zenodo that could limit us there, I advocated for RODARE because I know that we have a large quota and storage possibilities there.
